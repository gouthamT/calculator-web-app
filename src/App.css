.app {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: auto;
  display: flex;
  flex-direction: column;
  background: var(--bg-default);
}

.calculator {
  flex: 5;
}

/* media queries for non small devices */
@media (min-width: 768px) {
  .app {
    flex-direction: row;
  }
}