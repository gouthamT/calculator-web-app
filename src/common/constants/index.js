export const SCIENTIFIC = "scientific";
export const NUMBER = "number";
export const ARITHMETIC = "arithmetic";