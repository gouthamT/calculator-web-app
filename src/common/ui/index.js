export {
  SquareRootExpression,
  PowerExpression,
  LogarithmExpression,
} from './math-components';

export { ErrorBoundary } from './error-boundary';