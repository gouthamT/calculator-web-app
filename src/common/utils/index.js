import { handleClear, handleEqualOperator, handleNumberEntry, handlePercentageOperator, handleDotOperator } from './handlers';
import { compute } from "./compute-helper";

const isNumber = (item) => {
  return /[0-9]+/.test(item);
};

export const calculate = (obj, buttonName) => {
  try {

    if (buttonName === "AC") {
      return handleClear();
    }

    if (isNumber(buttonName)) {
      return handleNumberEntry(obj, buttonName);
    }

    if (buttonName === "%") {
      return handlePercentageOperator(obj)
    }

    if (buttonName === ".") {
      return handleDotOperator(obj);
    }

    if (["=", "Enter"].includes(buttonName)) {
      return handleEqualOperator(obj);
    }

    if (buttonName === "+/-") {
      if (obj.next) {
        return { next: (-1 * parseFloat(obj.next)).toString() };
      }
      if (obj.total) {
        return { total: (-1 * parseFloat(obj.total)).toString() };
      }
      return {};
    }

    // Button must be an operation
    if (obj.operation) {
      return {
        total: compute(obj.total, obj.next, obj.operation),
        next: null,
        operation: buttonName,
      };
    }

    // no operation yet, but the user typed one

    // The user hasn't typed a number yet, just save the operation
    if (!obj.next) {
      return { operation: buttonName };
    }

    // save the operation and shift 'next' into 'total'
    return {
      total: obj.next,
      next: null,
      operation: buttonName,
    };

  } catch {
    return handleClear();
  }
}
