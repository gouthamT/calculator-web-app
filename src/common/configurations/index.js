import React from 'react';
import { SquareRootExpression, PowerExpression, LogarithmExpression} from '../ui';

const ButtonConfigurationsMap = new Map(
  [
    ["(", 
      {
        key: "(",
        label: "(",
        operatorType: "scientific",
        className: "",
      }
    ],
    [")", 
      {
        key: ")",
        label: ")",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["mc", 
      {
        key: "mc",
        label: "mc",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["m+", 
      {
        key: "m+",
        label: "m+",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["m-",
      {
        key: "m-",
        label: "m-",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["mr",
      {
        key: "mr",
        label: "mr",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["AC",
      {
        key: "AC",
        label: "AC",
        className: "",
      }
    ],
    ["+/-", 
      {
        key: "+/-",
        label: "+/-",
        className: "",
      }
    ],
    ["%", 
      {
        key: "%",
        label: "%",
        className: "",
      }
    ],
    ["รท",
      {
        key: "รท",
        label: "รท",
        operatorType: "arithmetic",
        className: "",
      }
    ],
    ["second",
      {
        key: "second",
        label: <PowerExpression base="2" exponent="nd" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["xPow2",
      {
        key: "xPow2",
        label: <PowerExpression base="x" exponent="2" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["xPow3", 
      {
        key: "xPow3",
        label: <PowerExpression base="x" exponent="3" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["xPowY",
      {
        key: "xPowY",
        label: <PowerExpression base="x" exponent="y" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["ePowX",
      {
        key: "ePowX",
        label: <PowerExpression base="e" exponent="x" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["10PowX",
      {
        key: "10PowX",
        label: <PowerExpression base="10" exponent="x" />,
        operatorType: "scientific",
        className: "",
      }
    ],
    ["7",
      {
        key: "7",
        label: "7",
        operatorType: "number",
        className: "",
      }
    ],
    ["8", 
      {
        key: "8",
        label: "8",
        operatorType: "number",
        className: "",
      }
    ],
    ["9",
      {
        key: "9",
        label: "9",
        operatorType: "number",
        className: "",
      }
    ],
    ["x",
      {
        key: "x",
        label: "x",
        operatorType: "arithmetic",
        className: "",
      }
    ],
    ["1/x", 
      {
        key: "1/x",
        label: "1/x",
        operatorType: "scientific",
        className: "",
      }
    ],
    ["2SqrX", {
      key: "2SqrX",
      label: <SquareRootExpression radicand="X" index="2" />,
      operatorType: "scientific",
      className: "",
    }],
    ["3SqrX", {
      key: "3SqrX",
      label: <SquareRootExpression radicand="X" index="3" />,
      operatorType: "scientific",
      className: "",
    }],
    ["XSqrY", {
      key: "XSqrY",
      label: <SquareRootExpression radicand="Y" index="X" />,
      operatorType: "scientific",
      className: "",
    }],
    ["In", {
      key: "In",
      label: "In",
      operatorType: "scientific",
      className: "",
    }],
    ["log10", {
      key: "log10",
      label: <LogarithmExpression base="log" exponent="10" />,
      operatorType: "scientific",
      className: "",
    }],
    ["4", {
      key: "4",
      label: "4",
      operatorType: "number",
      className: "",
    }],
    ["5", {
      key: "5",
      label: "5",
      operatorType: "number",
      className: "",
    }],
    ["6", {
      key: "6",
      label: "6",
      operatorType: "number",
      className: "",
    }],
    ["-", {
      key: "-",
      label: "-",
      operatorType: "arithmetic",
      className: "",
    }],
    ["x!", {
      key: "x!",
      label: "x!",
      operatorType: "scientific",
      className: "",
    }],
    ["sin", {
      key: "sin",
      label: "sin",
      operatorType: "scientific",
      className: "",
    }],
    ["cos", {
      key: "cos",
      label: "cos",
      operatorType: "scientific",
      className: "",
    }],
    ["tan", {
      key: "tan",
      label: "tan",
      operatorType: "scientific",
      className: "",
    }],
    ["e", {
      key: "e",
      label: "e",
      operatorType: "scientific",
      className: "",
    }],
    ["EE", {
      key: "EE",
      label: "EE",
      operatorType: "scientific",
      className: "",
    }],
    ["1", {
      key: "1",
      label: "1",
      operatorType: "number",
      className: "",
    }],
    ["2", {
      key: "2",
      label: "2",
      operatorType: "number",
      className: "",
    }],
    ["3", {
      key: "3",
      label: "3",
      operatorType: "number",
      className: "",
    }],
    ["+", {
      key: "+",
      label: "+",
      operatorType: "arithmetic",
      className: "",
    }],
    ["Rad", {
      key: "Rad",
      label: "Rad",
      operatorType: "scientific",
      className: "",
    }],
    ["sinh",
    {
      key: "sinh",
      label: "sinh",
      operatorType: "scientific",
      className: "",
    }],
    ["cosh", {
      key: "cosh",
      label: "cosh",
      operatorType: "scientific",
      className: "",
    }],
    ["tanh", {
      key: "tanh",
      label: "tanh",
      operatorType: "scientific",
      className: "",
    }],
    ["nArray", {
      key: "nArray",
      label: <span>&#8719;</span>,
      operatorType: "scientific",
      className: "",
    }],
    ["Rand", {
      key: "Rand",
      label: "Rand",
      operatorType: "scientific",
      className: "small-font",
    }],
    ["0", {
      key: "0",
      label: "0",
      operatorType: "number",
      className: "expanded-button",
    }],
    [".", {
      key: ".",
      label: ".",
      operatorType: "number",
      className: "",
    }],
    ["=", {
      key: "=",
      label: "=",
      operatorType: "arithmetic",
      className: "",
    }],
]);

const ButtonConfigurations = Array.from(ButtonConfigurationsMap);

export { ButtonConfigurationsMap, ButtonConfigurations };