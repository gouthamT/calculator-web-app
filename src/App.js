import React from 'react';
import { useStore } from './controllers';
import { ErrorBoundary } from './common/ui';
import { ResultView, ButtonList, HistoryView } from './ui';
import './App.css';

const App = () =>  {
  const [displayValue] = useStore();

  return (
    <ErrorBoundary>
      <div className="app">
        <div className="calculator">
          <ResultView value={displayValue} />
          <ButtonList />
        </div>
        <HistoryView />
      </div>
    </ErrorBoundary>
  );
}

export default App;
