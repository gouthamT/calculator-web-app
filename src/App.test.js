import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ButtonConfigurationsMap } from './common/configurations';
import { act } from 'react-dom/test-utils';
import App from './App';

test('should match all configured buttons', () => {
  const { getByTestId } = render(<App />);

  expect(ButtonConfigurationsMap.size).toBe(49);
  ButtonConfigurationsMap.forEach((_, key) => {
    expect(getByTestId(`btn-${key}`)).toMatchSnapshot();
  });
});

test('should have results view', () => {
  const { getByTestId } = render(<App />);

  expect(getByTestId(`input-box`).value).toBe("0");
});

test('should have history view', () => {
  const { findByTestId } = render(<App />);

  expect(findByTestId(`histories`)).not.toBeNull();
});

test('should perform addition operation and update state', async () => {
  const { getByTestId } = render(<App />);
  await act(async () => {
    fireEvent.click(getByTestId('btn-3'))
    fireEvent.click(getByTestId('btn-+'))
    fireEvent.click(getByTestId('btn-5'))
    fireEvent.click(getByTestId('btn-='))
  })

  expect(getByTestId(`input-box`).value).toBe("8");
});

test('should perform subtraction operation and update state', async () => {
  const { getByTestId } = render(<App />);
  await act(async () => {
    fireEvent.click(getByTestId('btn-8'))
    fireEvent.click(getByTestId('btn--'))
    fireEvent.click(getByTestId('btn-5'))
    fireEvent.click(getByTestId('btn-='))
  })

  expect(getByTestId(`input-box`).value).toBe("3");
});

test('should perform multiplication operation and update state', async () => {
  const { getByTestId } = render(<App />);
  await act(async () => {
    fireEvent.click(getByTestId('btn-8'))
    fireEvent.click(getByTestId('btn-x'))
    fireEvent.click(getByTestId('btn-5'))
    fireEvent.click(getByTestId('btn-='))
  })

  expect(getByTestId(`input-box`).value).toBe("40");
});

test('should perform division operation and update state', async () => {
  const { getByTestId } = render(<App />);
  await act(async () => {
    fireEvent.click(getByTestId('btn-2'))
    fireEvent.click(getByTestId('btn-0'))
    fireEvent.click(getByTestId('btn-0'))
    fireEvent.click(getByTestId('btn-รท'))
    fireEvent.click(getByTestId('btn-5'))
    fireEvent.click(getByTestId('btn-='))
  })

  expect(getByTestId(`input-box`).value).toBe("40");
});

test('should perform multiple operations and update state', async () => {
  const { getByTestId } = render(<App />);
  await act(async () => {
    fireEvent.click(getByTestId('btn-1'))
    fireEvent.click(getByTestId('btn-+'))
    fireEvent.click(getByTestId('btn-2'))
    fireEvent.click(getByTestId('btn-x'))
    fireEvent.click(getByTestId('btn-3'))
    fireEvent.click(getByTestId('btn-รท'))
    fireEvent.click(getByTestId('btn-4'))
    fireEvent.click(getByTestId('btn-='))
  })

  expect(getByTestId(`input-box`).value).toBe("2.25");
});