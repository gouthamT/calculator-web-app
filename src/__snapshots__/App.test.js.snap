// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match all configured buttons 1`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="("
  data-testid="btn-("
>
  (
</button>
`;

exports[`should match all configured buttons 2`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key=")"
  data-testid="btn-)"
>
  )
</button>
`;

exports[`should match all configured buttons 3`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="mc"
  data-testid="btn-mc"
>
  mc
</button>
`;

exports[`should match all configured buttons 4`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="m+"
  data-testid="btn-m+"
>
  m+
</button>
`;

exports[`should match all configured buttons 5`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="m-"
  data-testid="btn-m-"
>
  m-
</button>
`;

exports[`should match all configured buttons 6`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="mr"
  data-testid="btn-mr"
>
  mr
</button>
`;

exports[`should match all configured buttons 7`] = `
<button
  class="button actionable-button bg-grey "
  data-operator-key="AC"
  data-testid="btn-AC"
>
  AC
</button>
`;

exports[`should match all configured buttons 8`] = `
<button
  class="button actionable-button bg-grey "
  data-operator-key="+/-"
  data-testid="btn-+/-"
>
  +/-
</button>
`;

exports[`should match all configured buttons 9`] = `
<button
  class="button actionable-button bg-grey "
  data-operator-key="%"
  data-testid="btn-%"
>
  %
</button>
`;

exports[`should match all configured buttons 10`] = `
<button
  class="button actionable-button pure-white bg-orange "
  data-operator-key="÷"
  data-testid="btn-÷"
>
  ÷
</button>
`;

exports[`should match all configured buttons 11`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="second"
  data-testid="btn-second"
>
  <div
    class="power"
  >
    <span>
      2
    </span>
    <sup
      class="power__suffix"
    >
      nd
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 12`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="xPow2"
  data-testid="btn-xPow2"
>
  <div
    class="power"
  >
    <span>
      x
    </span>
    <sup
      class="power__suffix"
    >
      2
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 13`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="xPow3"
  data-testid="btn-xPow3"
>
  <div
    class="power"
  >
    <span>
      x
    </span>
    <sup
      class="power__suffix"
    >
      3
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 14`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="xPowY"
  data-testid="btn-xPowY"
>
  <div
    class="power"
  >
    <span>
      x
    </span>
    <sup
      class="power__suffix"
    >
      y
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 15`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="ePowX"
  data-testid="btn-ePowX"
>
  <div
    class="power"
  >
    <span>
      e
    </span>
    <sup
      class="power__suffix"
    >
      x
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 16`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="10PowX"
  data-testid="btn-10PowX"
>
  <div
    class="power"
  >
    <span>
      10
    </span>
    <sup
      class="power__suffix"
    >
      x
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 17`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="7"
  data-testid="btn-7"
>
  7
</button>
`;

exports[`should match all configured buttons 18`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="8"
  data-testid="btn-8"
>
  8
</button>
`;

exports[`should match all configured buttons 19`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="9"
  data-testid="btn-9"
>
  9
</button>
`;

exports[`should match all configured buttons 20`] = `
<button
  class="button actionable-button pure-white bg-orange "
  data-operator-key="x"
  data-testid="btn-x"
>
  x
</button>
`;

exports[`should match all configured buttons 21`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="1/x"
  data-testid="btn-1/x"
>
  1/x
</button>
`;

exports[`should match all configured buttons 22`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="2SqrX"
  data-testid="btn-2SqrX"
>
  <div
    class="square-root"
  >
    <sup
      class="square-root__prefix"
    >
      2
    </sup>
    <span>
      √
    </span>
    <span
      class="square-root__base"
    >
      X
    </span>
  </div>
</button>
`;

exports[`should match all configured buttons 23`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="3SqrX"
  data-testid="btn-3SqrX"
>
  <div
    class="square-root"
  >
    <sup
      class="square-root__prefix"
    >
      3
    </sup>
    <span>
      √
    </span>
    <span
      class="square-root__base"
    >
      X
    </span>
  </div>
</button>
`;

exports[`should match all configured buttons 24`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="XSqrY"
  data-testid="btn-XSqrY"
>
  <div
    class="square-root"
  >
    <sup
      class="square-root__prefix"
    >
      X
    </sup>
    <span>
      √
    </span>
    <span
      class="square-root__base"
    >
      Y
    </span>
  </div>
</button>
`;

exports[`should match all configured buttons 25`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="In"
  data-testid="btn-In"
>
  In
</button>
`;

exports[`should match all configured buttons 26`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="log10"
  data-testid="btn-log10"
>
  <div
    class="logarithm"
  >
    <span>
      log
    </span>
    <sup
      class="logarithm__suffix"
    >
      10
    </sup>
  </div>
</button>
`;

exports[`should match all configured buttons 27`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="4"
  data-testid="btn-4"
>
  4
</button>
`;

exports[`should match all configured buttons 28`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="5"
  data-testid="btn-5"
>
  5
</button>
`;

exports[`should match all configured buttons 29`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="6"
  data-testid="btn-6"
>
  6
</button>
`;

exports[`should match all configured buttons 30`] = `
<button
  class="button actionable-button pure-white bg-orange "
  data-operator-key="-"
  data-testid="btn--"
>
  -
</button>
`;

exports[`should match all configured buttons 31`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="x!"
  data-testid="btn-x!"
>
  x!
</button>
`;

exports[`should match all configured buttons 32`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="sin"
  data-testid="btn-sin"
>
  sin
</button>
`;

exports[`should match all configured buttons 33`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="cos"
  data-testid="btn-cos"
>
  cos
</button>
`;

exports[`should match all configured buttons 34`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="tan"
  data-testid="btn-tan"
>
  tan
</button>
`;

exports[`should match all configured buttons 35`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="e"
  data-testid="btn-e"
>
  e
</button>
`;

exports[`should match all configured buttons 36`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="EE"
  data-testid="btn-EE"
>
  EE
</button>
`;

exports[`should match all configured buttons 37`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="1"
  data-testid="btn-1"
>
  1
</button>
`;

exports[`should match all configured buttons 38`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="2"
  data-testid="btn-2"
>
  2
</button>
`;

exports[`should match all configured buttons 39`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="3"
  data-testid="btn-3"
>
  3
</button>
`;

exports[`should match all configured buttons 40`] = `
<button
  class="button actionable-button pure-white bg-orange "
  data-operator-key="+"
  data-testid="btn-+"
>
  +
</button>
`;

exports[`should match all configured buttons 41`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="Rad"
  data-testid="btn-Rad"
>
  Rad
</button>
`;

exports[`should match all configured buttons 42`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="sinh"
  data-testid="btn-sinh"
>
  sinh
</button>
`;

exports[`should match all configured buttons 43`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="cosh"
  data-testid="btn-cosh"
>
  cosh
</button>
`;

exports[`should match all configured buttons 44`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="tanh"
  data-testid="btn-tanh"
>
  tanh
</button>
`;

exports[`should match all configured buttons 45`] = `
<button
  class="button scientific white bg-dark-grey "
  data-operator-key="nArray"
  data-testid="btn-nArray"
>
  <span>
    ∏
  </span>
</button>
`;

exports[`should match all configured buttons 46`] = `
<button
  class="button scientific white bg-dark-grey small-font"
  data-operator-key="Rand"
  data-testid="btn-Rand"
>
  Rand
</button>
`;

exports[`should match all configured buttons 47`] = `
<button
  class="button actionable-button white bg-light-grey expanded-button"
  data-operator-key="0"
  data-testid="btn-0"
>
  0
</button>
`;

exports[`should match all configured buttons 48`] = `
<button
  class="button actionable-button white bg-light-grey "
  data-operator-key="."
  data-testid="btn-."
>
  .
</button>
`;

exports[`should match all configured buttons 49`] = `
<button
  class="button actionable-button pure-white bg-orange "
  data-operator-key="="
  data-testid="btn-="
>
  =
</button>
`;
